### ISSUE TITLE
Is it possible to take screenshots if more than 1 instance of the browsers are running?

### ISSUE THREAD

**Issue by arjunjh10** [rluba/mocha-proshot#1]
My question is around testSharding and the way we can use the mocha-proshot reporter to be able to take screenshots for the failing specs.

I am trying to experiment sharding in my test suite by launching 2 instances of the chrome browser and enabling testSharding. But when my tests fail on either instances of chrome, the reporter fails to store a screenshot.

**Comment by rluba** [comment]
Interesting problem! I haven't used sharding with protractor yet, so I barely guess what the issue might be. Can you provide an example project with failing tests?

**Comment by arjunjh10** [comment]
So, an example of how I have used sharding in my project is as follows
My protractor.conf file is set up like this for sharding purposes:

```
process.env.PROSHOT_DIR = './protractor-screenshots'
process.env.multi = 'spec=- mocha-proshot=-'
.....
....
 capabilities: {
    'browserName': 'chrome',
    shardTestFiles: true,
    maxInstances: 2 <could be anything, in my project I am experimenting with only two instances initially>
  },
```

And here are the settings in my protractor-jenkins.conf file

```
process.env.PROSHOT_DIR = './protractor-screenshots'
process.env.multi = 'spec=- mocha-proshot=-'

extend(true, config, {
  // Port that the jenkins selenium hub is running on.
  seleniumAddress: <address>,

  mochaOpts: {
    reporter: 'spec', <It was failing with mocha-multi on 2 instances of chrome so I had to change it to spec reporter here>
    enableTimeouts: true,
    timeout: 30000
  }
})
```

Now, normally what would happen when I run my tests in Jenkins and locally with only 1 browser instance is that for any failed tests the screenshots get saved in the `./protractor-screenshots` folder both locally and on the jenkins within the build info.

But when I experimented my framework with sharding and used the `mocha-proshot` expecting to capture screenshots then it failed both locally and on jenkins. 
In my view, it does not know which browser is to be considered for taking the screenshot as there are 2 in my case.

Here is a snapshot of my test report from the console log in Jenkins

```
17:20:44 [launcher] 0 instance(s) of WebDriver still running
17:20:44 [launcher] chrome #1a passed
17:20:44 [launcher] chrome #1c passed
17:20:44 [launcher] chrome #1d passed
17:20:44 [launcher] chrome #1b passed
17:20:44 [launcher] chrome #1f passed
17:20:44 [launcher] chrome #1g passed
17:20:44 [launcher] chrome #1e passed
17:20:44 [launcher] chrome #1i failed 1 test(s)
17:20:44 [launcher] chrome #1k passed
17:20:44 [launcher] chrome #1l passed
17:20:44 [launcher] chrome #1j failed 4 test(s)
17:20:44 [launcher] chrome #1h passed
17:20:44 [launcher] chrome #1m passed
17:20:44 [launcher] chrome #1n passed
17:20:44 [launcher] chrome #1p passed
17:20:44 [launcher] chrome #1r passed
17:20:44 [launcher] chrome #1s passed
17:20:44 [launcher] chrome #1y failed 2 test(s)
17:20:44 [launcher] chrome #1{ passed
17:20:44 [launcher] chrome #1w failed 4 test(s)
17:20:44 [launcher] chrome #1} passed
17:20:44 [launcher] chrome #1z failed 8 test(s)
17:20:44 [launcher] chrome #1| failed 8 test(s)
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1~ passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 passed
17:20:44 [launcher] chrome #1 failed 1 test(s)
17:20:44 [launcher] chrome #1 failed 1 test(s)
17:20:44 [launcher] chrome #1 failed 1 test(s)
17:20:44 [launcher] chrome #1 failed 3 test(s)
17:20:44 [launcher] overall: 33 failed spec(s)
```

I also found this in one of the failed tests in jenkins, maybe it is of some help, dont know.

```
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, chrome={userDataDir=/tmp/.com.google.Chrome.76N8Jr}, takesHeapSnapshot=true, databaseEnabled=false, handlesAlerts=true, version=43.0.2357.132, platform=LINUX, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
```

Please let me know if this info helps in any way :)

**Comment by arjunjh10** [comment]
Hi,
Following up on this one.

**Comment by rluba** [comment]
Hi @arjunjh10,

I finally got around to testing `mocha-proshot` + sharding.

The problem is that Protractor [expects an `err` property on each failed test](https://github.com/angular/protractor/blob/master/lib/frameworks/mocha.js#L92). But this property is not set by mocha itself – it only exists due to [a side-effect of using a builtin reporter](https://github.com/mochajs/mocha/blob/master/lib/reporters/base.js#L272).
- The good news: I can add a workaround for this to `mocha-proshot`.
- The bad news: `mocha-multi` has the same problem and therefore doesn't work together with Protractor's sharding. So you have to use `mocha-proshot` without `mocha-multi` if you want to enable sharding.

**Comment by rluba** [comment]
Fixed in [1.0.1](https://github.com/rluba/mocha-proshot/releases/tag/v1.0.1).
